Makefile

```diff
@@ -13,13 +13,21 @@ default: $(DEFAULT_TARGET)
 VERBOSE ?= 0
 
 CHPL := chpl
+
+# We need to make the HDF5 API use the 1.10.x version for compatibility between 1.10 and 1.12
+CHPL_HDF5_COMPAT := -DH5_USE_110_API
+
+# We need to build up compiler flags for Chapel --ccflags arg because they go in quotes
+CHPL_CC_FLAGS := $(CHPL_HDF5_COMPAT)
```

Comment:
Note that `--ccflags` stacks, so instead of building up all the args you could use it multiple times. e.g. instead of `--ccflags="-O1 -DH5_USE_110_API"` you could instead do `--ccflags="-O1" ccflags="-DH5_USE_110_API"`

That would probably make this diff smaller since I _think_ all you would need is `CHPL_FLAGS += --ccflags="-DH5_USE_110_API"`, but I'm not positive.

Reply:
Ah, I wasn't sure about the stacking that's good to know.  I was having issues with the quoted string making it over to the chapel based unit tests.  I'm going to see if I can do the stacking thing without making the escape-quoting stuff go crazy.

Reply:
I'm not sure that the quotes are even needed with stacking. I can't remember why I included them in the first place, it should only be needed if your have multiple options separated with a space.